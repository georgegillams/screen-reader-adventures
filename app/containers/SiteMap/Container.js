import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { cssModules } from 'gg-components/helpers/cssModules';
import { SubSection, TextLink, PageTitle } from 'gg-components/Typography';
import { LoadingIndicator } from 'gg-components/LoadingIndicator';

import { NON_EMOJI_REGEX } from 'helpers/regexConstants';
import redirects from 'helpers/redirects';
import STYLES from 'containers/pages.scss';

const getClassName = cssModules(STYLES);

export default class SiteMap extends React.Component {
  componentWillMount = () => {
    this.props.loadBlogs();
  };

  render() {
    const { blogsLoadError, blogs, className } = this.props;
    const outerClassNameFinal = [];

    if (className) {
      outerClassNameFinal.push(className);
    }

    return (
      <div className={outerClassNameFinal.join(' ')}>
        <Helmet title="SiteMap" />
        <LoadingIndicator loading={!blogs} error={blogsLoadError}>
          <PageTitle style={{ lineHeight: '1.5rem' }} name="Site map">
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Blog 📝"
            >
              {blogs &&
                blogs.map(blog => {
                  if (!blog.showInBlogsList) {
                    return null;
                  }
                  return (
                    <div>
                      <TextLink href={`/blog/${blog.id}`}>
                        {blog.title
                          ? blog.title.match(NON_EMOJI_REGEX).join('')
                          : 'no name'}
                      </TextLink>
                      <br />
                    </div>
                  );
                })}
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Travel ✈️"
            >
              {blogs &&
                blogs.map(blog => {
                  if (!blog.showInTravelBlogsList) {
                    return null;
                  }
                  return (
                    <div>
                      <TextLink href={`/travel/${blog.id}`}>
                        {blog.title
                          ? blog.title.match(NON_EMOJI_REGEX).join('')
                          : 'no name'}
                      </TextLink>
                      <br />
                    </div>
                  );
                })}
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Other blogs"
            >
              {blogs &&
                blogs.map(blog => {
                  if (blog.showInTravelBlogsList || blog.showInBlogsList) {
                    return null;
                  }
                  return (
                    <div>
                      <TextLink href={`/blog/${blog.id}`}>
                        {blog.title
                          ? blog.title.match(NON_EMOJI_REGEX).join('')
                          : 'no name'}
                      </TextLink>
                      <br />
                    </div>
                  );
                })}
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Photography 🎨"
            >
              <TextLink href="/photography">
                See some artistic creations
              </TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Work 📱"
            >
              <TextLink href="/work">Overview</TextLink>
              <br />
              <TextLink href="/work/backpack">Backpack</TextLink>
              <br />
              <TextLink href="/work/degree">Degree</TextLink>
              <br />
              <TextLink href="/work/epicc">EPICC</TextLink>
              <br />
              <TextLink href="/work/side-projects">Side projects</TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Machine learning 🤖"
            >
              <TextLink href="/ml/grammar">
                Using ML to predict grammatical correctness.
              </TextLink>
            </SubSection>
            {/* <SubSection anchor={false} className={getClassName('pages__site-map-item')} name="Documents 🥇">
        <TextLink  href="/documents/degree">
         <SubSection anchor={false} nclassName={getClassName('pages__site-map-item')}oPadding link name="Degree Certificate - 2018
      </TextLink>
        <br />
    </SubSection> */}
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Design 🎨"
            >
              <TextLink href="/design/privacy-policy">Privacy Policy</TextLink>
              <br />
              <TextLink href="/design/colours">Colours</TextLink>
              <br />
              <TextLink href="/design/components">Components</TextLink>
              <br />
              <TextLink href="/design/typography">Typography</TextLink>
              <br />
              <TextLink href="/design/lab">Lab</TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Other stuff 🤷‍♂️"
            >
              <TextLink href="/about">About me</TextLink>
              <br />
              <TextLink href="/contact">Contact</TextLink>
              <br />
              <TextLink href="/payments">Payments</TextLink>
              <br />
              <TextLink href="/monzoPots">Monzo savings tracking</TextLink>
              <br />
              <TextLink href="/support">Support</TextLink>
              <br />
              <TextLink href="/status">Status</TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Random 🐉"
            >
              <TextLink external href="/robots.txt">
                Robots.txt{' '}
              </TextLink>
              <br />
              <TextLink external href="/sitemap.xml">
                SiteMap.xml{' '}
              </TextLink>
              <br />
              <TextLink href="/page-not-found"> 404 Page</TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="API"
            >
              <TextLink href="/api-docs">API docs</TextLink>
              <br />
              <TextLink external href="/greasemonkey/find_backpack_components">
                /greasemonkey/find_backpack_components{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_travis_new_tab">
                /greasemonkey/github_travis_new_tab{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_squash_reminder">
                /greasemonkey/github_squash_reminder{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/gurushots_boost">
                /greasemonkey/gurushots_boost{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/secureEcs_download">
                /greasemonkey/secureEcs_download{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/skyscanner_buttons">
                /greasemonkey/skyscanner_buttons{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_highlight_name">
                /greasemonkey/github_highlight_name{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_expand_comments">
                /greasemonkey/github_expand_comments{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/hackthis_coding_1">
                /greasemonkey/hackthis_coding_1{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/hackthis_coding_2">
                /greasemonkey/hackthis_coding_2{' '}
              </TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Accounts 🔑"
            >
              <TextLink href="/sign-up">Sign up</TextLink>
              <br />
              <TextLink href="/account">Account</TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Admin 👮‍♂️"
            >
              <TextLink href="/admin/users">Users</TextLink>
              <br />
              <TextLink href="/admin/analytics">Analytics</TextLink>
              <br />
              <TextLink href="/monzoPots">Monzo</TextLink>
              <br />
              <TextLink href="/admin/notifications">Notifications</TextLink>
              <br />
              <TextLink href="/admin/blog">Blogs</TextLink>
              <br />
              <TextLink href="/payments">Payments</TextLink>
            </SubSection>
            <SubSection
              anchor={false}
              className={getClassName('pages__site-map-item')}
              name="Redirects 👉"
            >
              {redirects.map(redirect => (
                <div>
                  <TextLink href={`${redirect.from}`}>
                    {`${redirect.from} ⇒ ${redirect.to}`}
                  </TextLink>
                  <br />
                </div>
              ))}
            </SubSection>
          </PageTitle>
        </LoadingIndicator>
      </div>
    );
  }
}

SiteMap.propTypes = {
  blogsLoadError: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  blogs: PropTypes.object,
  filter: PropTypes.func,
  linkPrefix: PropTypes.string,
  loadBlogs: PropTypes.func.isRequired,
  className: PropTypes.string,
};
