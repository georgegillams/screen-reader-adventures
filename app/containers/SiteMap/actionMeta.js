const actionMeta = {
  key: 'sitemap',
  actionDefinitions: [
    {
      LOAD_BLOGS: 'tbd',
      attributes: [],
      stateMutations: {
        loadingBlogs: true,
        loadBlogsError: null,
      },
    },
    {
      LOAD_BLOGS_REGISTER_SUCCESS: 'tbd',
      attributes: ['blogs'],
      stateMutations: {
        loadingBlogs: false,
        loadBlogsSuccess: true,
        blogs: action => action.blogs,
      },
    },
    {
      LOAD_BLOGS_REGISTER_ERROR: 'tbd',
      attributes: ['loadBlogsError'],
      stateMutations: {
        loadingBlogs: false,
        loadBlogsSuccess: false,
        loadBlogsError: action => action.loadBlogsError,
      },
    },
  ],
};

export default actionMeta;
