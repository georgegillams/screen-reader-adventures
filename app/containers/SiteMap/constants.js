export const LOAD_BLOGS = 'boilerplate/SiteMap/LOAD_BLOGS';
export const LOAD_BLOGS_ERROR = 'boilerplate/SiteMap/LOAD_BLOGS';
export const LOAD_BLOGS_SUCCESS = 'boilerplate/SiteMap/LOAD_BLOGS_SUCCESS';
