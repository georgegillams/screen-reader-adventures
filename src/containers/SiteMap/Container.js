import React from 'react';
import { Subsection } from 'gg-components/Subsection';
import TextLink from 'components/common/TextLink';
import PageTitle from 'components/common/PageTitle';
import { Paragraph } from 'gg-components/Paragraph';
import redirects from 'helpers/redirects';
import STYLES from './site-map.scss';
import { cssModules } from 'gg-components/helpers/cssModules';

const getClassName = cssModules(STYLES);

const SiteMap = props => (
  <PageTitle name="Site map" {...props}>
    <Subsection anchor={false} name="Design 🎨" className={getClassName('site-map__section')}>
      <Paragraph>
        <TextLink href="/privacy-policy">Privacy Policy</TextLink>
      </Paragraph>
    </Subsection>
    <Subsection anchor={false} name="Other stuff 🤷‍♂️" className={getClassName('site-map__section')}>
      <Paragraph>
        <TextLink href="/debug">Debug tools</TextLink>
        <br />
        <TextLink href="/status">Status</TextLink>
      </Paragraph>
    </Subsection>
    <Subsection anchor={false} name="Random 🐉" className={getClassName('site-map__section')}>
      <Paragraph>
        <TextLink hrefExternal href="/robots.txt">
          Robots.txt
        </TextLink>
        <br />
        <TextLink hrefExternal href="/sitemap.xml">
          SiteMap.xml
        </TextLink>
        <br />
        <TextLink href="/404">404 error page - not found</TextLink>
        <br />
        <TextLink href="/teapot">418 error page - I&apos;m a teapot</TextLink>
      </Paragraph>
    </Subsection>
    <Subsection anchor={false} name="API" className={getClassName('site-map__section')}>
      <Paragraph>
        <TextLink href="/api-docs">API docs</TextLink>
      </Paragraph>
    </Subsection>
    <Subsection anchor={false} name="Accounts 🔑" className={getClassName('site-map__section')}>
      <Paragraph>
        <TextLink href="/account">Account</TextLink>
        <br />
        <TextLink href="/sign-up">Sign up</TextLink>
        <br />
        <TextLink href="/login">Log in</TextLink>
        <br />
        <TextLink href="/email-verification">Email verification</TextLink>
        <br />
        <TextLink href="/magic-login">Magic login</TextLink>
      </Paragraph>
    </Subsection>
    <Subsection anchor={false} name="Admin 👮‍♂️" className={getClassName('site-map__section')}>
      <Paragraph>
        <TextLink href="/admin">Admin navigation</TextLink>
        <br />
        <TextLink href="/admin/analytics">Analytics</TextLink>
        <br />
        <TextLink href="/admin/emails">Emails</TextLink>
        <br />
        <TextLink href="/admin/notifications">Notifications</TextLink>
        <br />
        <TextLink href="/admin/users">Users</TextLink>
      </Paragraph>
    </Subsection>
    <Subsection anchor={false} name="Redirects" className={getClassName('site-map__section')}>
      <Paragraph>
        {redirects.map(redirect => (
          <div key={redirect.from}>
            <TextLink href={`${redirect.from}`}>{`${redirect.from} ⇒ ${redirect.to}`}</TextLink>
            <br />
          </div>
        ))}
      </Paragraph>
    </Subsection>
  </PageTitle>
);

export default SiteMap;
