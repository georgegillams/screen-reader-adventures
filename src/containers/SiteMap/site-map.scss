@import '~utils/tokens';

.site-map {
  &__section {
    word-break: break-word;
  }
}
