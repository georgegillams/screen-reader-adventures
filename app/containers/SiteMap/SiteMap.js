import React from 'react';
import PropTypes from 'prop-types';
import { Helmet } from 'react-helmet';
import { SubSection, TextLink } from 'components/Typography';
import LoadingIndicator from 'components/LoadingIndicator';
import { NON_EMOJI_REGEX } from 'helpers/constants';
import redirects from 'helpers/redirects';
import STYLES from 'containers/pages.scss';
import { cssModules } from 'bpk-react-utils';
const getClassName = cssModules(STYLES); // REGEX_REPLACED

export default class SiteMap extends React.Component {
  componentWillMount = () => {
    this.props.loadBlogs();
  };

  render() {
    const { loading, error, blogs, loadBlogs, className, ...rest } = this.props;
    const outerClassNameFinal = [
      getClassName('pages__container'),
      getClassName('pages__container--centered'),
    ];

    if (className) {
      outerClassNameFinal.push(className);
    }

    return (
      <div className={outerClassNameFinal.join(' ')} {...rest}>
        <Helmet title="SiteMap" />
        <LoadingIndicator loading={loading} error={error}>
          <div>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Blog 📝"
            >
              {blogs &&
                blogs.map(blog => {
                  if (!blog.showInBlogsList) {
                    return null;
                  }
                  return (
                    <div>
                      <TextLink href={`/blog/${blog.id}`}>
                        {blog.title
                          ? blog.title.match(NON_EMOJI_REGEX).join('')
                          : 'no name'}
                      </TextLink>
                      <br />
                    </div>
                  );
                })}
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Travel ✈️"
            >
              {blogs &&
                blogs.map(blog => {
                  if (!blog.showInTravelBlogsList) {
                    return null;
                  }
                  return (
                    <div>
                      <TextLink href={`/travel/${blog.id}`}>
                        {blog.title
                          ? blog.title.match(NON_EMOJI_REGEX).join('')
                          : 'no name'}
                      </TextLink>
                      <br />
                    </div>
                  );
                })}
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Other blogs"
            >
              {blogs &&
                blogs.map(blog => {
                  if (blog.showInTravelBlogsList || blog.showInBlogsList) {
                    return null;
                  }
                  return (
                    <div>
                      <TextLink href={`/blog/${blog.id}`}>
                        {blog.title
                          ? blog.title.match(NON_EMOJI_REGEX).join('')
                          : 'no name'}
                      </TextLink>
                      <br />
                    </div>
                  );
                })}
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Photography 🎨"
            >
              <TextLink href="/photography">
                See some artistic creations
              </TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Work 📱"
            >
              <TextLink href="/work/degree">Degree</TextLink>
              <br />
              <TextLink href="/work">Portfolio</TextLink>
              <br />
              <TextLink href="/work/bpk-component-demo">
                Backpack Demo Component
              </TextLink>
              <br />
              <TextLink href="/apps/password-character-extractor">
                Password Character Extractor
              </TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Machine learning 🤖"
            >
              <TextLink href="/ml/grammar">
                Using ML to predict grammatical correctness.
              </TextLink>
            </SubSection>
            {/* <SubSection noAnchor className={getClassName('pages__site-map-item')} name="Documents 🥇">
        <TextLink  href="/documents/degree">
         <SubSection noAnchor nclassName={getClassName('pages__site-map-item')}oPadding link name="Degree Certificate - 2018
      </TextLink>
        <br />
    </SubSection> */}
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Design 🎨"
            >
              <TextLink href="/design/privacy-policy">Privacy Policy</TextLink>
              <br />
              <TextLink href="/design/colours">Colours</TextLink>
              <br />
              <TextLink href="/design/components">Components</TextLink>
              <br />
              <TextLink href="/design/typography">Typography</TextLink>
              <br />
              <TextLink href="/design/lab">Lab</TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Other stuff 🤷‍♂️"
            >
              <TextLink href="/about">About me</TextLink>
              <br />
              <TextLink href="/contact">Contact</TextLink>
              <br />
              <TextLink href="/gts">Location tracking</TextLink>
              <br />
              <TextLink href="/payments">Payments</TextLink>
              <br />
              <TextLink href="/monzoPots">Monzo savings</TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Random 🐉"
            >
              <TextLink external href="/robots.txt">
                Robots.txt{' '}
              </TextLink>
              <br />
              <TextLink external href="/sitemap.xml">
                SiteMap.xml{' '}
              </TextLink>
              <br />
              <TextLink href="/418"> Error 418: I&apos;m a teapot</TextLink>
              <br />
              <TextLink href="/page-not-found"> 404 Page</TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="API"
            >
              <TextLink external href="/api/gts/loadLatest">
                Tracking API endpoint{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/find_backpack_components">
                /greasemonkey/find_backpack_components{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_travis_new_tab">
                /greasemonkey/github_travis_new_tab{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_squash_reminder">
                /greasemonkey/github_squash_reminder{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/gurushots_boost">
                /greasemonkey/gurushots_boost{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/secureEcs_download">
                /greasemonkey/secureEcs_download{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/skyscanner_buttons">
                /greasemonkey/skyscanner_buttons{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_highlight_name">
                /greasemonkey/github_highlight_name{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/github_expand_comments">
                /greasemonkey/github_expand_comments{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/hackthis_coding_1">
                /greasemonkey/hackthis_coding_1{' '}
              </TextLink>
              <br />
              <TextLink external href="/greasemonkey/hackthis_coding_2">
                /greasemonkey/hackthis_coding_2{' '}
              </TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Accounts 🔑"
            >
              <TextLink href="/sign-up">Sign up</TextLink>
              <br />
              <TextLink href="/account">Account</TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Admin 👮‍♂️"
            >
              <TextLink href="/admin/sessions">Sessions</TextLink>
              <br />
              <TextLink href="/admin/blogs">Blogs</TextLink>
              <br />
              <TextLink href="/admin/blog-comments">Blog comments</TextLink>
              <br />
              <TextLink href="/admin/notifications">Notifications</TextLink>
              <br />
              <TextLink href="/admin/payments">Payments</TextLink>
              <br />
              <TextLink href="/admin/ping-pen-testing">
                Ping pen-testing
              </TextLink>
            </SubSection>
            <SubSection
              noAnchor
              className={getClassName('pages__site-map-item')}
              name="Redirects 👉"
            >
              {redirects.map(redirect => (
                <div>
                  <TextLink href={`${redirect.from}`}>
                    {`${redirect.from} ⇒ ${redirect.to}`}
                  </TextLink>
                  <br />
                </div>
              ))}
            </SubSection>
          </div>
        </LoadingIndicator>
      </div>
    );
  }
}

SiteMap.propTypes = {
  loading: PropTypes.bool,
  error: PropTypes.oneOfType([PropTypes.object, PropTypes.bool]),
  blogs: PropTypes.object,
  filter: PropTypes.func,
  linkPrefix: PropTypes.string,
  loadBlogs: PropTypes.func.isRequired,
  className: PropTypes.string,
};
