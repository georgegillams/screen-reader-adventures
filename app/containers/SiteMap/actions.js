import { LOAD_BLOGS, LOAD_BLOGS_SUCCESS, LOAD_BLOGS_ERROR } from './constants';

export function loadBlogs() {
  return {
    type: LOAD_BLOGS,
  };
}

export function loadBlogsSuccess(blogs) {
  return {
    type: LOAD_BLOGS_SUCCESS,
    blogs,
  };
}

export function loadBlogsError(error) {
  return {
    type: LOAD_BLOGS_ERROR,
    error,
  };
}
