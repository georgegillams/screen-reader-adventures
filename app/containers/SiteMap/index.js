import { connect } from 'react-redux';
import { compose } from 'redux';
import { createStructuredSelector } from 'reselect';
import injectReducer from 'utils/injectReducer';
import injectSaga from 'utils/injectSaga';
import {
  makeSelectBlogs,
  makeSelectLoading,
  makeSelectError,
} from './selectors';
import { loadBlogs } from './actions';
import reducer from './reducer';
import saga from './saga';
import SiteMap from './SiteMap';

const mapDispatchToProps = dispatch => ({
  loadBlogs: () => dispatch(loadBlogs()),
});

const mapStateToProps = createStructuredSelector({
  blogs: makeSelectBlogs(),
  loading: makeSelectLoading(),
  error: makeSelectError(),
});

const withConnect = connect(
  mapStateToProps,
  mapDispatchToProps,
);

const withReducer = injectReducer({ key: 'site-map', reducer });
const withSaga = injectSaga({ key: 'site-map', saga });

export default compose(
  withReducer,
  withSaga,
  withConnect,
)(SiteMap);
export { mapDispatchToProps };
