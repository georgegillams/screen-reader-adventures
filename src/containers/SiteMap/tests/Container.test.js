import React from 'react';
import { render } from '@testing-library/react';

import SiteMap from '../Container';

describe('<SiteMap />', () => {
  it('should render correctly', () => {
    const { container } = render(<SiteMap />);

    expect(container).toMatchSnapshot();
  });
});
