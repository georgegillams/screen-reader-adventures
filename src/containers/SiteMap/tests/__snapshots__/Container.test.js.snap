// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteMap /> should render correctly 1`] = `
<div>
  <h1
    class="page-title__heading"
  >
    Site map
  </h1>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="design-🎨"
    >
      Design 🎨
    </h3>
    <span
      class="paragraph"
    >
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/privacy-policy"
      >
        Privacy Policy
      </a>
    </span>
  </div>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="other-stuff-🤷‍♂️"
    >
      Other stuff 🤷‍♂️
    </h3>
    <span
      class="paragraph"
    >
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/debug"
      >
        Debug tools
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/status"
      >
        Status
      </a>
    </span>
  </div>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="random-🐉"
    >
      Random 🐉
    </h3>
    <span
      class="paragraph"
    >
      <a
        aria-label="Robots.txt (opens in a new tab)"
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/robots.txt"
        rel="noopener noreferrer"
        target="_blank"
      >
        Robots.txt
        <div
          class="text-link__icon-container"
        >
          <svg
            class="icon text-link__icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
            />
          </svg>
        </div>
      </a>
      <br />
      <a
        aria-label="SiteMap.xml (opens in a new tab)"
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/sitemap.xml"
        rel="noopener noreferrer"
        target="_blank"
      >
        SiteMap.xml
        <div
          class="text-link__icon-container"
        >
          <svg
            class="icon text-link__icon"
            height="18"
            viewBox="0 0 24 24"
            width="18"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
            />
          </svg>
        </div>
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/404"
      >
        404 error page - not found
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/teapot"
      >
        418 error page - I'm a teapot
      </a>
    </span>
  </div>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="api"
    >
      API
    </h3>
    <span
      class="paragraph"
    >
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/api-docs"
      >
        API docs
      </a>
    </span>
  </div>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="accounts-🔑"
    >
      Accounts 🔑
    </h3>
    <span
      class="paragraph"
    >
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/account"
      >
        Account
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/sign-up"
      >
        Sign up
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/login"
      >
        Log in
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/email-verification"
      >
        Email verification
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/magic-login"
      >
        Magic login
      </a>
    </span>
  </div>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="admin-👮‍♂️"
    >
      Admin 👮‍♂️
    </h3>
    <span
      class="paragraph"
    >
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/admin"
      >
        Admin navigation
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/admin/analytics"
      >
        Analytics
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/admin/emails"
      >
        Emails
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/admin/notifications"
      >
        Notifications
      </a>
      <br />
      <a
        class="text-link text-link--text-link text-link--no-padding text-link--inline"
        href="/admin/users"
      >
        Users
      </a>
    </span>
  </div>
  <div
    class="subsection__outer site-map__section"
  >
    <h3
      class="subsection__heading"
      id="redirects"
    >
      Redirects
    </h3>
    <span
      class="paragraph"
    >
      <div>
        <a
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="/codebase"
        >
          /codebase ⇒ https://github.com/georgegillams/screen-reader-adventures
        </a>
        <br />
      </div>
      <div>
        <a
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="/wp-admin"
        >
          /wp-admin ⇒ /admin
        </a>
        <br />
      </div>
      <div>
        <a
          class="text-link text-link--text-link text-link--no-padding text-link--inline"
          href="/site-map"
        >
          /site-map ⇒ /sitemap
        </a>
        <br />
      </div>
    </span>
  </div>
</div>
`;
